src/Response/Product.php

```diff
@@ -1023,4 +1139,390 @@ protected function processTags(array $data)
 
         return $this;
     }
+
+    /**
+     * Set results, only should be used for testing and mocking
+     *
+     * @param array $data
+     * @return $this
+     */
+    public function setResults(array $data)
+    {
+        $this->results = $data;
+
+        return $this;
+    }
+
+    /**
+     * Get method name which is missing some data and pass the message to log class
+     *
+     * @param string $additionalInfo
+     * @return $this
+     */
+    protected function handleEmptyData($additionalInfo = '')
+    {
+        /**
+         * @var \Logger $log
+         */
+        if ($this->registry && ($log = $this->registry->get('log'))) {
+            $method = debug_backtrace();
+            $method = $method[1]['function'];
+            $message = 'Class ' . get_class($this) . ' method: ' . $method . ' is missing some data.';
+
+            if ($additionalInfo) {
+                $message .= ' Class message: ' . $additionalInfo;
+            }
+
+            $log->trace($message);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param string $lang
+     * @return $this
+     */
+    public function setLanguageCode($lang)
+    {
+        $this->languageCode = strtoupper($lang);
+
+        return $this;
+    }
+
+    /**
+     * @codeCoverageIgnore - Ignore this method as it used for better mocking
+     */
+    public function getLanguageCode()
+    {
+        return $this->languageCode;
+    }
+
+    /**
+     * @param string $country
+     * @return $this
+     */
+    public function setTaxRateCountryCode($country)
+    {
+        $this->countryCode = strtoupper($country);
+
+        return $this;
+    }
+
+    /**
+     * @codeCoverageIgnore - Ignore this method as it used for better mocking
+     */
+    public function getTaxRateCountryCode()
+    {
+        return $this->countryCode;
+    }
+
+    /**
+     * @param string $url
+     * @return $this
+     */
+    public function setStoreUrl($url)
+    {
+        $this->storeUrl = rtrim($url, '/');
+
+        return $this;
+    }
+
+    /**
+     * @codeCoverageIgnore - Ignore this method as it used for better mocking
+     */
+    public function getStoreUrl()
+    {
+        return $this->storeUrl;
+    }
+
+    /**
+     * @param int $storePlentyId
+     * @return $this
+     */
+    public function setStorePlentyId($storePlentyId)
+    {
+        $this->storePlentyId = $storePlentyId;
+
+        return $this;
+    }
+
+    /**
+     * @codeCoverageIgnore - Ignore this method as it used for better mocking
+     */
+    public function getStorePlentyId()
+    {
+        return $this->storePlentyId;
+    }
+
+    /**
+     * @codeCoverageIgnore - Ignore this method as it used for better mocking
+     */
+    public function getDefaultEmptyValue()
+    {
+        return '';
+    }
+
+    /**
```

Comment:
Getters for Wrapper to call. 
If $this->fields[''] is used, it is placeholders for the future, when variations data will be streamed as well. Because for these getters variation data is required. 
Also found some old fields, getters for them returns null now. I do not seem to find code that would retrieve those values, so I am keeping the getters for them.